module-2/backend/week_one.md

```diff
@@ -5,28 +5,116 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+
+  *Instance variables*
+  
+6. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
   
   ```ruby
   get '/horses' do
     erb :index
   end
   ```
 
-8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed`?
-9. What's the purpose of ERB?
-10. Why do I need a development AND test database?
-11. What's responsive design?
-12. What is CRUD and why is it important?
-13. What does HTTP stand for? 
-14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
-15. What's an ORM?
-16. What's the most commonly used ORM?
-17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
-18. What's a migration? 
-19. When you create a migration, does it automatically modify your database?
-20. How does a model relate to a database?
+
+*Add @count = 1 inside your get '/horses/ loop*
+
+
+7. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed`?
+
+  *update the row erb :index to be erb :index, :locals=> {:name => "Mr. Ed"}*
+
+8. What's the purpose of ERB?
+
+   *It allows us to use ruby code in our HTML view*
+
+  
+9. Why do I need a development AND test database?
+
+    *You need both so that you can test without it impacting your database*
+
+10. What's responsive design?
+
+    *Design that adapts to different screen sizes*
+  
+11. What is CRUD and why is it important?
+
+    *CRUD stands for create, read, update and delete.  
+    It's important because it represents the basic functions of an application that interacts with a database*
+    
+12. What does HTTP stand for? 
+
+    *Hyper Text transfer protocol*
+    
+13. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+
+  *<%= ruby code %> prints out the code
+  <% ruby code %> runs code but doesn't print it*
+
+14. What's an ORM?
+
+  *Object relationship mapping*
+
+15. What's the most commonly used ORM?  
+
+  *Active record?*
+
+16. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+ ```
+  get '/restaurants' do   - read
+    renders a view of all the restaurants 
+  
+  get '/restaurants/:id' do   - read
+    renders a view of a single restaurant
+  
+  get '/restaurants/new' do  - create
+    renders a new erb template 
+  
+  post 'restaurants' do  - create
+    redirects to the view of restaurants after the user has created a new one
+  
+  get '/restaurants/:id/edit' do - update
+    renders an edit view
+  
+  post 'restaurants/:id' do -  update

```

Comment:
don't forget the / at the beginning of the pathname here too! also - this should be a `put`. we use `put` when we update an existing record and we use `post` when we're creating an entirely new record 👍